src/us/kbase/workspace/database/provenance/CreditMetadata.java

```diff
@@ -0,0 +1,465 @@
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import us.kbase.workspace.database.Util;
+/*
+ * A class representing the credit metadata for a workspace object.
+ */
+public class CreditMetadata {
+
+	public enum ResourceType {
+		DATASET;
+
+		/**
+		 * Get a resource type based on a supplied string.
+		 *
+		 * @param str
+		 *                a string representing a resource type.
+		 * @return a resource type.
+		 * @throws IllegalArgumentException
+		 *                 if there is no resource type
+		 *                 related to the input string.
+		 */
+		public static ResourceType getResourceType(final String resourceType) {
+			final String lowercaseInput = Util.checkString(resourceType, "resourceType")
+					.toLowerCase();
+			switch (lowercaseInput) {
+				case "dataset":
+				case "data_set":
+				case "data set":
+					return ResourceType.DATASET;
+				default:
+					throw new IllegalArgumentException(
+							"Invalid resourceType: " + resourceType);
+			}
+		}
+	}
+
+	private final String identifier;
+	private final String license;
+	private final String version;
+	private final ResourceType resourceType;
+	private final List<String> comments;
+	private final List<Contributor> contributors;
+	private final List<EventDate> dates;
+	private final List<FundingReference> funding;
+	private final List<PermanentID> relatedIdentifiers;
+	private final List<Title> titles;
+
+	private static final ResourceType DEFAULT_RESOURCE_TYPE = ResourceType.DATASET;
+
+	private CreditMetadata(
+			final String identifier,
+			final String license,
+			final String version,
+			final ResourceType resourceType,
+			final List<String> comments,
+			final List<Contributor> contributors,
+			final List<EventDate> dates,
+			final List<FundingReference> funding,
+			final List<PermanentID> relatedIdentifiers,
+			final List<Title> titles) {
+		this.comments = comments;
+		this.contributors = contributors;
+		this.dates = dates;
+		this.funding = funding;
+		this.identifier = identifier;
+		this.license = license;
+		this.relatedIdentifiers = relatedIdentifiers;
+		this.resourceType = resourceType;
+		this.titles = titles;
+		this.version = version;
+	}
+
+	/**
+	 * Gets the identifier for the resource.
+	 *
+	 * @return identifier
+	 */
+	public String getIdentifier() {
+		return identifier;
+	}
+
+	/*
+	 * Gets the license for the resource; may be a string like 'Apache 2.0' or an URL in string
+	 * form.
+	 *
+	 * @return resource license (if present)
+	 */
+	public Optional<String> getLicense() {
+		return Optional.ofNullable(license);
+	}
+
+	/*
+	 * Gets the version information for the resource.
+	 *
+	 * @return resource version (if present)
+	 */
+	public Optional<String> getVersion() {
+		return Optional.ofNullable(version);
+	}
+
+	/*
+	 * Gets the resource type.
+	 *
+	 * @return resource type as a {@link ResourceType}
```

Comment:
This is totally fine, but for future reference the java docs will automatically link to the definition of the return type so the link is redundant

Reply:
So what should I put in the future for this kind of thing?

```java
 *
 * @return the resource type as a ResourceType
 */
 ```
 
 or just
 ```java
 *
 * @return the resource type
 */

Reply:
The latter is what I do. The getters and setters we're dealing with in a value class are usually so simple I hardly thing javadocs are necessary, but it's considered bad form so...